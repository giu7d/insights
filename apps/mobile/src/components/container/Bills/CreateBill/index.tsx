import { Image, Text, TouchableOpacity, View } from 'react-native'

import FeatherIcon from '@expo/vector-icons/Feather'
import colors from 'tailwindcss/colors'

import MastercardLogoImage from '@assets/mastercard-logo.png'

export default function CreateBill() {
  return (
    <>
      {/* Payment Information */}
      <View className="flex-row justify-between items-center py-2">
        <View className="w-12 h-12 rounded-full bg-gray-200" />
        <Text className="flex-grow mx-4 font-semibold text-lg text-gray-800">
          Alex Lee
        </Text>
        <Text className="font-semibold text-lg text-gray-800">- $90.00</Text>
      </View>

      {/* Divider */}
      <View className="border-b-2 border-black w-full opacity-5 my-4" />

      {/* Information Label/Value */}
      <View className="py-4 gap-2">
        <Text className="font-semibold text-md text-gray-500">Status</Text>
        <View>
          <View className="flex-row items-center gap-2">
            <View className="bg-green-500 w-6 h-6 items-center justify-center rounded-full">
              <FeatherIcon name="check" color={colors.white} />
            </View>
            <Text className="font-semibold text-md text-gray-800">
              Successful
            </Text>
          </View>
        </View>
      </View>

      {/* Information Label/Value */}
      <View className="py-4 gap-2">
        <Text className="font-semibold text-md text-gray-500">Date</Text>
        <Text className="font-semibold text-md text-gray-800">
          Aug 30, 2022 at 8:36pm
        </Text>
      </View>

      {/* Information Label/Value */}
      <View className="py-4 gap-2">
        <Text className="font-semibold text-md text-gray-500">
          Payment method
        </Text>
        <View className="p-4 flex-row items-center bg-gray-100 rounded-2xl">
          <View className="h-12 w-12 bg-gray-700 rounded-full items-center justify-center">
            <Image source={MastercardLogoImage} className="h-5 w-8 -scale-75" />
          </View>
          <View className="flex-grow px-4">
            <Text className="font-semibold text-sm text-gray-900">
              Mastercard
            </Text>
            <Text className="text-sm text-gray-500">Ending in 9999</Text>
          </View>
        </View>
      </View>

      {/* Button Area */}
      <View className="py-4 gap-2 flex-row">
        {/* Button MD */}
        <TouchableOpacity className="bg-blue-100 justify-center items-center p-4 rounded-xl flex-grow">
          <Text className="text-md font-semibold text-blue-600">Invoice</Text>
        </TouchableOpacity>
        {/* Button MD */}
        <TouchableOpacity className="bg-gray-800 justify-center items-center p-4 rounded-xl flex-grow">
          <Text className="text-md font-semibold text-white">Send again</Text>
        </TouchableOpacity>
      </View>
    </>
  )
}
