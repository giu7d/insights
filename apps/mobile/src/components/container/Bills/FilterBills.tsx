import { useState } from 'react'

import Tab from '@/components/fragments/Tab'

const FILTER_ITEMS = [
  { name: 'all', label: 'All' },
  { name: 'unpaid', label: 'Unpaid' },
  { name: 'paid', label: 'Paid' }
]

export default function FilterBills() {
  const [selected, setSelected] = useState('all')

  return <Tab items={FILTER_ITEMS} selected={selected} onSelect={setSelected} />
}
