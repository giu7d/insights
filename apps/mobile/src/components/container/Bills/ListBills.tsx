import { FlatList, View } from 'react-native'

import { Feather } from '@expo/vector-icons'
import { HoldItem } from 'react-native-hold-menu'
import colors from 'tailwindcss/colors'

import BillCard from '@/components/fragments/Card/BillCard'
import { trpc } from '@/services/api'

const MENU_ITEMS = [
  {
    text: 'Open',
    icon: () => (
      <Feather name="corner-up-right" color={colors.white} size={18} />
    ),
    onPress: () => {}
  },
  {
    text: 'Share',
    icon: () => <Feather name="share" color={colors.white} size={18} />,
    onPress: () => {}
  },
  {
    text: 'Delete',
    icon: () => <Feather name="trash" color={colors.red[500]} size={18} />,
    isDestructive: true,
    onPress: () => {}
  }
]

export default function ListBills() {
  const { data } = trpc.bills.list.useQuery()

  if (!data) return <></>

  return (
    <FlatList
      horizontal
      showsHorizontalScrollIndicator={false}
      data={data}
      contentContainerStyle={{ padding: 24 }}
      ItemSeparatorComponent={() => <View className="w-4" />}
      renderItem={({ item }) => (
        <View key={`bill-${item.name}`}>
          <HoldItem items={MENU_ITEMS} hapticFeedback="Medium">
            <BillCard bill={item} />
          </HoldItem>
        </View>
      )}
    />
  )
}
