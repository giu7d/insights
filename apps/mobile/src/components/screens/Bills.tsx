import { View } from 'react-native'

import CreateBill from '@/components/container/Bills/CreateBill'
import FilterBills from '@/components/container/Bills/FilterBills'
import ListBills from '@/components/container/Bills/ListBills'
import MainHeader from '@/components/container/Header/Main'
import MainTab from '@/components/container/Tab/Main'
import DrawerTemplate from '@/components/templates/Drawer'
import MainTemplate from '@/components/templates/Main'
import useDrawer from '@/hooks/useDrawer'
import { trpc } from '@/services/api'

export default function Bills() {
  const drawer = useDrawer()
  const billsQuery = trpc.bills.list.useQuery()

  const handleRefetchBills = async () => {
    await new Promise((resolve) =>
      setTimeout(() => billsQuery.refetch().finally(() => resolve('')), 1000)
    )
  }

  return (
    <View testID="bill-screen">
      <DrawerTemplate renderDrawer={<CreateBill />}>
        <MainTemplate
          onRefresh={handleRefetchBills}
          renderHeader={
            <MainHeader>
              <View className="p-6 pt-4">
                <FilterBills />
              </View>
            </MainHeader>
          }
          renderFooter={<MainTab onPressNew={drawer.show} />}
        >
          <View className="flex-grow h-full pb-20">
            <ListBills />
          </View>
        </MainTemplate>
      </DrawerTemplate>
    </View>
  )
}
