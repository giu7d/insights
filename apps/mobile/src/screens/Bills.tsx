import { View } from 'react-native'

import CreateBill from '@/components/container/Bills/CreateBill'
import FilterBills from '@/components/container/Bills/FilterBills'
import ListBills from '@/components/container/Bills/ListBills'
import Header from '@/components/container/Header'
import PrimaryButton from '@/components/fragments/Button/Primary'
import {
  LinearOpacityBottomContainer,
  LinearOpacityTopContainer
} from '@/components/fragments/Containers/LinearOpacity'
import BaseTemplate from '@/components/templates/Base'
import DrawerTemplate from '@/components/templates/Drawer'
import useDrawer from '@/hooks/useDrawer'

export default function Bills() {
  const drawer = useDrawer()

  return (
    <DrawerTemplate drawerComponent={<CreateBill />}>
      <BaseTemplate
        renderHeader={(isCompact) => (
          <LinearOpacityTopContainer>
            <Header isCompact={isCompact} />
            <View className="p-6 pt-4">
              <FilterBills />
            </View>
          </LinearOpacityTopContainer>
        )}
        renderFooter={() => (
          <LinearOpacityBottomContainer className="absolute w-full p-6 pt-9 bottom-0">
            <PrimaryButton onPress={drawer.show}>Create new bill</PrimaryButton>
          </LinearOpacityBottomContainer>
        )}
        scrollViewProps={{
          className: 'gap-6'
        }}
      >
        <View className="px-6 pb-12">
          <ListBills />
        </View>
      </BaseTemplate>
    </DrawerTemplate>
  )
}
