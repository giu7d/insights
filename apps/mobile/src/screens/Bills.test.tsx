import renderWithProvider from '@/config/render'
import Bills from '@/screens/Bills'

describe('<Bills />', () => {
  it('renders correctly', () => {
    const component = renderWithProvider(<Bills />)

    expect(component.toJSON()).toMatchSnapshot()
  })
})
