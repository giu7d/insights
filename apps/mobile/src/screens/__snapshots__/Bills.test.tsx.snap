// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bills /> renders correctly 1`] = `null`;
